### DISCUSSION TITLE
Using `any_()` with `contains()` on a JSONB column

### DISCUSSION THREAD

**Discussion by jmriebold** [sqlalchemy/sqlalchemy#9824] [Usage Questions]
I'm trying to query for rows where a JSONB column contains any of an array of JSON values. The raw SQL query I wrote works (cribbing from [this post](https://dba.stackexchange.com/a/130863)), and looks like this:

```sql
SELECT field_a, field_b
FROM table_a
WHERE field_b @> ANY (CAST(ARRAY['[{"foo": "bar"}]', '[{"foo": "baz"}]'] AS JSONB[]))
```

which i translated into SQLA as

```python
stmt = stmt.filter(table_a.field_b.contains(any_(cast(array([str([{'foo': ii}]) for ii in ['bar', 'baz']]), ARRAY(JSONB)))))
```

However, that produces the following SQL, which results in a syntax error:

```sql
SELECT field_a, field_b
FROM table_a
WHERE field_b @> (ANY (CAST(ARRAY['[{"foo": "bar"}]', '[{"foo": "baz"}]'] AS JSONB[])))
```

It's almost exactly what I want, but the extra parentheses break the query.

**Comment by CaselIT** [comment]
Hi,

where you able to fix this?

**Reply by CaselIT** [reply]
ok I see that an issue was opened

**Reply by jmriebold** [reply]
Yeah, just converted to an issue since I'm inclined to think it's a bug at this point.
